doc/nrf/app_dev/device_guides/nrf54l/cryptography.rst

```diff
@@ -0,0 +1,300 @@
+
+From a programmer's point of view, keys stored in the KMU or generated by the IKG are referenced using built-in keys that are key IDs in the range from ``MBEDTLS_PSA_KEY_ID_BUILTIN_MIN`` to and inclusive ``MBEDTLS_PSA_KEY_ID_BUILTIN_MAX``.
+
+KMU
+***
+
+The nRF54L Series KMU can store cryptographic keys and a 384-bit random seed in key storage slots.
+The nrf_cracen PSA driver exposes the KMU operations through standard PSA Crypto API calls, with some vendor-specific extensions.
+Supported KMU operations include:
+
+* Importing/provisioning keys to KMU slots
+* Deleting a key from the KMU, allowing the underlying storage location to be re-used
+* Revoking a key from the KMU, preventing re-use of the underlying key slots
+* Directly pushing symmetric keys to CRACEN's engine for symmetric crypto operations, without exposing the key material to CPU
+* Directly pushing a 384-bit seed to CRACEN's IKG to derive isolated keys
+* Pushing asymmetric keys to CPU RAM before loading them into CRACEN's engine for asymmetric operations
+
+Additionally, the nrf_cracen PSA driver supports storing encrypted keys in KMU slots, transparently decrypting them to a temporary RAM location before using them in cryptographic operations.
+
+.. _ug_nrf54l_crypto_kmu_slots:
+
+KMU slots
+=========
+
+The KMU is partitioned into 256 numbered slots, each capable of storing 128 bits of key material, a 32-bit target push address and 32 bits of metadata.
+Storing keys that are larger than 128 bits is supported by using multiple, consecutive slots.
+
+The application can use KMU slots to store key data for their own purposes.
+Some KMU slots are reserved for current and future |NCS| use cases.
+
+.. list-table:: List of reserved KMU slots
+   :widths: auto
+   :header-rows: 1
+
+   * - | Reserved KMU slots
+       | (range inclusive)
+     - NCS usage
+     - Description
+   * - 180-182
+     - Reserved
+     - --
+   * - 183-185
+     - IKG seed
+     - 384-bit random seed to generate keys using the CRACEN IKG
+   * - 186-225
+     - Reserved
+     - --
+   * - 226-227
+     - UROT_PUBKEY_0
+     - | Revokable MCUboot firmware image key, generation 0.
+       | ED25519 public key.
+   * - 228-229
+     - UROT_PUBKEY_1
+     - | Revokable MCUboot firmware image key, generation 1.
+       | ED25519 public key.
+   * - 230-231
+     - UROT_PUBKEY_2
+     - | Revokable MCUboot firmware image key, generation 2.
+       | ED25519 public key.
+   * - 232-241
+     - Reserved
+     - --
+   * - 242-243
+     - BL_PUBKEY_0
+     - | Revokable NSIB firmware image key, generation 0.
+       | ED25519 public key.
+   * - 244-245
+     - BL_PUBKEY_1
+     - | Revokable NSIB firmware image key, generation 1.
+       | ED25519 public key.
+   * - 246-247
+     - BL_PUBKEY_2
+     - | Revokable NSIB firmware image key, generation 2.
+       | ED25519 public key.
+   * - 248-255
+     - Reserved
+     - --
+
+Programming model for referencing keys
```

Comment:
In this section, we should mention https://arm-software.github.io/psa-api/ or https://arm-software.github.io/psa-api/crypto/1.2/, so users know which psa functions we refer to in the table.